src/app/admin/edit-product/edit-product.component.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { BehaviorSubject, of, Subject } from 'rxjs';
+import { switchMap, takeUntil } from 'rxjs/operators';
+
+import { Product } from '../../products/product.interface';
+import { ProductsService } from '../../products/products.service';
+import { NotificationService } from '../../core/notification.service';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Component({
+  selector: 'app-edit-product',
+  templateUrl: './edit-product.component.html',
+  styleUrls: ['./edit-product.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class EditProductComponent implements OnInit, OnDestroy {
+  form$: BehaviorSubject<FormGroup | null> =
+    new BehaviorSubject<FormGroup | null>(null);
+  product: Product | null = null;
+  requestInProgress = false;
+
+  private readonly onDestroy$: Subject<void> = new Subject();
+
+  constructor(
+    private readonly activatedRoute: ActivatedRoute,
+    private readonly fb: FormBuilder,
+    private readonly notificationService: NotificationService,
+    private readonly productsService: ProductsService,
+    private readonly router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.activatedRoute.paramMap
+      .pipe(
+        switchMap((paramMap) =>
+          paramMap.has('productId')
+            ? this.productsService.getProductById(paramMap.get('productId')!)
+            : of(null)
+        ),
+        takeUntil(this.onDestroy$)
+      )
+      .subscribe((product) => {
+        this.product = product;
+        this.buildForm();
+      });
+  }
```

Comment:
Could use the snapshot as there are no navigation when only the route param changes

Reply:
Fixed